### PR TITLE
changes to removed required

### PR DESCRIPTION
**Pull request by randymgeorge**
## Describe your changes

- ...

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: api/kessel/inventory/v1beta1/k8s_clusters_service.proto [hunk 1]
```diff
@@ -15,10 +15,7 @@ message CreateK8sClusterRequest {
   K8sCluster k8s_cluster = 1;
 }
 
-message CreateK8sClusterResponse {
-  // The K8s cluster created in Kessel Asset Inventory
-  K8sCluster k8s_cluster = 1;
-}
+message CreateK8sClusterResponse {}
```
Comment (josejulio) [inline api/kessel/inventory/v1beta1/k8s_clusters_service.proto:18]: Asked in the previous PR:

About the requirement of not returning anything in the create requests.
Wouldn't this be an issue for the caller if they want to do any subsequent call with the newly created resource?
e.g. getting the id of the newly created resource for any following up process to initialize or prepare anything else?

File: api/kessel/inventory/v1beta1/notifications_integration.proto [hunk 2]
```diff
@@ -14,7 +14,7 @@ option java_package = "org.project_kessel.api.inventory.v1beta1";
 
 message NotificationsIntegration {
   // Metadata about this resource
-  Metadata metadata = 1 [ (google.api.field_behavior) = REQUIRED, (validate.rules).any.required = true ];
+  Metadata metadata = 1;
```
Comment (josejulio) [inline api/kessel/inventory/v1beta1/notifications_integration.proto:17]:
```suggestion
  Metadata metadata = 1 [ (validate.rules).any.required = true ];
```
We will still require the required annotation. It generates code to do the validation.

File: api/kessel/inventory/v1beta1/k8s_clusters_service.proto (hunk 1 above)
Comment (randymgeorge) [reply api/kessel/inventory/v1beta1/k8s_clusters_service.proto:18]: If you look at the update/delete interfaces, they leverage the reporter information, not the "internal" ID.  Thus, they do not need to get the ID returned.

File: api/kessel/inventory/v1beta1/notifications_integration.proto (hunk 2 above)
Comment (randymgeorge) [reply api/kessel/inventory/v1beta1/notifications_integration.proto:17]: metadata should not be required.  There are no fields that are required for one to pass in.

File: api/kessel/inventory/v1beta1/policy.proto [hunk 3]
```diff
@@ -14,16 +12,10 @@ option java_package = "org.project_kessel.api.inventory.v1beta1";
...
-      [ (google.api.field_behavior) = INPUT_ONLY ];
-
-  // The entities that registered this item in the Kessel Asset Inventory. The
-  // same resource may be registered by multiple reporters.
-  repeated ReporterData reporters = 353323086
-      [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  ReporterData reporter_data = 245278792;
 
-  PolicyDetail data = 2122698;
+   PolicyDetail data = 2122698;
```
Comment (josejulio) [inline api/kessel/inventory/v1beta1/policy.proto:20]: nitpick: looks like one space slip in there. Surprised our linter didn't catch it. I guess it doesn't care about protobuf files.

```suggestion
  PolicyDetail data = 2122698;
```

File: api/kessel/inventory/v1beta1/rhel_hosts_service.proto [hunk 4]
```diff
@@ -13,38 +12,28 @@ option java_package = "org.project_kessel.api.inventory.v1beta1";
...
-  ];
+  RhelHost host = 1;
 }
 
 message CreateRhelHostResponse {
   // The Rhel Host created in Kessel Asset Inventory
-  RhelHost host = 1 [
-    (google.api.field_behavior) = REQUIRED
-  ];
+  RhelHost host = 1;
```
Comment (josejulio) [inline api/kessel/inventory/v1beta1/rhel_hosts_service.proto:20]: IIUC this host shouldn't be here as we don't return anything in the response. Is that right?
```suggestion
```

Comment (randymgeorge) [reply api/kessel/inventory/v1beta1/rhel_hosts_service.proto:20]: I did not change the hosted resources return as part of this PR.  There are some other changes that need to happen for that to be removed

Comment (josejulio) [reply api/kessel/inventory/v1beta1/rhel_hosts_service.proto:20]: Sounds good, I can take care of it in a follow up

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**